### PR TITLE
http2: updates method signatures to remove nghttp2-specific names and types.

### PR DESCRIPTION
**Pull request by birenroy**
Signed-off-by: Biren Roy <birenroy@google.com>

Commit Message: Removes nghttp2-specific names and types from CodecImpl. Cleanup related to https://github.com/envoyproxy/envoy/issues/23596.
Additional Description:
Risk Level: low
Testing: ran http and integration tests locally
Docs Changes:
Release Notes:
Platform Specific Features:

### CODE REVIEW

File: test/common/http/http2/protocol_constraints_test.cc [hunk 1]
```diff
@@ -161,13 +161,14 @@ TEST_F(ProtocolConstraintsTest, Priority) {
...
   constraints.incrementOpenedStreamCount();
 
-  nghttp2_frame_hd frame;
-  frame.type = NGHTTP2_PRIORITY;
-  EXPECT_TRUE(constraints.trackInboundFrames(&frame, 0).ok());
-  EXPECT_TRUE(constraints.trackInboundFrames(&frame, 0).ok());
-  EXPECT_TRUE(constraints.trackInboundFrames(&frame, 0).ok());
-  EXPECT_TRUE(constraints.trackInboundFrames(&frame, 0).ok());
-  EXPECT_TRUE(isBufferFloodError(constraints.trackInboundFrames(&frame, 0)));
+  size_t length = 5;
```
Comment (diannahu) [inline test/common/http/http2/protocol_constraints_test.cc:164]: no action: Huh, I hadn't realized that nghttp2 filled in length/flags for frame types where those are constant. (I guess this is a piece of info I can forget soon enough :P)

**Review by diannahu — APPROVED** [review]
Nice clean-up!

File: source/common/http/http2/codec_impl.cc [hunk 2]
```diff
@@ -1980,7 +1983,8 @@ Status ClientConnectionImpl::onBeginHeaders(const nghttp2_frame* frame) {
   RELEASE_ASSERT(frame->headers.cat == NGHTTP2_HCAT_RESPONSE ||
                      frame->headers.cat == NGHTTP2_HCAT_HEADERS,
                  "");
-  RETURN_IF_ERROR(trackInboundFrames(&frame->hd, frame->headers.padlen));
+  RETURN_IF_ERROR(trackInboundFrames(frame->hd.stream_id, frame->hd.length, frame->hd.type,
```
Comment (RyanTheOptimist) [inline source/common/http/http2/codec_impl.cc:1986]: Will this type (nghttp2_frame) be changing in a future PR? (Same question about the enum values like NGHTTP2_HEADERS)

**Review by RyanTheOptimist — APPROVED** [review]
Looks great. one question.

File: source/common/http/http2/codec_impl.cc (hunk 2 above)
Comment (birenroy) [reply source/common/http/http2/codec_impl.cc:1986]: Yes, that's the plan. I'll tag these PRs with https://github.com/envoyproxy/envoy/issues/23596.

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**